os/arch/arm/src/s5j/s5j_i2s.c

```diff
@@ -1600,47 +1600,47 @@ static int i2s_pause(struct i2s_dev_s *dev)
 {
 	struct s5j_i2s_s *priv = (struct s5j_i2s_s *)dev;
 	irqstate_t flags;
+	volatile u32 reg = 0;
 	DEBUGASSERT(priv);
 
+	reg = getreg32(priv->base + S5J_I2S_CON);
+	flags = irqsave();
 #ifdef I2S_HAVE_TX_P
-	if (priv->txpenab) {
-		flags = irqsave();
+	if (priv->txpenab && reg & I2S_CR_TXDMACTIVE) {
 		modifyreg32(priv->base + S5J_I2S_CON, 0, I2S_CR_TXCHPAUSE);
-		irqrestore(flags);
 	}
 #endif
 
 #ifdef I2S_HAVE_RX
-	if (priv->rxenab) {
-		flags = irqsave();
+	if (priv->rxenab && reg & I2S_CR_RXDMACTIVE) {
 		modifyreg32(priv->base + S5J_I2S_CON, 0, I2S_CR_RXCHPAUSE);
-		irqrestore(flags);
 	}
 #endif
+	irqrestore(flags);
 	return 0;
```

Comment:
you don't have to explicitly save and restore the irq state when you use  modifyreg32. It internally does it. take care of this in all places

Reply:
I want both rx and tx channels to pause / resume in one shot. 
So, I think  its better to disable irq explicitly.. Please let me know your opinion.

Reply:
But both features are mutually exclusive. In a single codec either one will be active at any given point of time. So oneshot enable disable doesn't make sense.

Reply:
Yes true. But we had not planned for it. 
In that case, we will have to modify all our API's to handle tx and rx separately. 
I will do the changes and publish them.

Reply:
Fixed.